bin/cheribsdtest/cheribsdtest.c

```diff
@@ -945,6 +945,45 @@ static const struct cheri_test cheri_tests[] = {
 	  .ct_desc = "Check bounds on a 1,048,576-byte dynamic stack allocation",
 	  .ct_func = test_bounds_stack_dynamic_1048576, },
 
+	/*
+	 * Test bounds on varargs.
+	 */
+	{ .ct_name = "test_bounds_varargs_vaarg_overflow",
+	  .ct_desc = "check that va_arg() triggers a fault on overrun",
+	  .ct_func = test_bounds_varargs_vaarg_overflow,
+	  .ct_flags = CT_FLAG_SIGNAL | CT_FLAG_SI_CODE | CT_FLAG_SI_TRAPNO,
+	  .ct_signum = SIGPROT,
+	  .ct_si_code = PROT_CHERI_BOUNDS,
+	  .ct_si_trapno = TRAPNO_CHERI,
+#ifndef __CHERI_PURE_CAPABILITY__
```

Comment:
I feel these should not even be built and run for hybrid, they just don't make sense there. XFAIL is more for things that should pass but currently don't due to a lack of engineering effort, not things that could never work. The existing globals/stack bounds tests are a bit of a strange case, since as written they will never work for hybrid, though with `__capability &` we can have them somewhat work for hybrid (only somewhat because we can't pad symbols in other translation units, so it's questionable whether we should also pad when in the same translation unit as that would be inconsistent, but technically more secure).